crates/rover-config/src/lib.rs

```diff
@@ -0,0 +1,67 @@
+mod layout;
+
+use anyhow::{Error, Result};
+use serde::{Deserialize, Serialize};
+use std::collections::HashMap;
+use std::env;
+use std::fs;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Config {
+    profiles: HashMap<String, Profile>,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct Profile {
+    api_key: String,
+}
+
+pub fn get_api_key(profile: Option<String>) -> Result<String> {
+    match env::var("APOLLO_API_KEY").ok() {
+        Some(api_key) => Ok(api_key),
+        None => {
+            let p = match profile {
+                Some(p) => p,
+                None => String::from("default"),
```

Comment:
NBD: Generally speaking I like constant strings used in multiple places to be defined once, although this is contained to this one file, so 🤷 TIOLI

Reply:
what does TIOLI mean?

Reply:
take it or leave it :D

Reply:
(also i agree we could make this a const 👍 )